### DISCUSSION TITLE
Github Management

### DISCUSSION THREAD

**Discussion by MatthewBozin** [Together-100Devs/Together#47] [General]
As the project grows and takes on new contributors, it would be very useful to overhaul the devops side of the github repo/project. This will improve clarity for new contributors, especially newer devs. It could also reduce the amount of onboarding the project leads would have to do.

Here are some proposed features:

- Remove tasks from the [readme](https://github.com/Caleb-Cohen/Together/blob/main/Readme.md) and make the [project kanban board](https://github.com/users/Caleb-Cohen/projects/1/views/1) the single source of truth for tasks
- Add [issue/pull request templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/manually-creating-a-single-issue-template-for-your-repository) that include contribution guidelines and self-check reminders. [Example PR template](https://github.com/MatthewBozin/Caesura/blob/main/.github/pull_request_template.md)
- Creating new issue labels to denote frontend/backend/design/devops issues
- Allow contributors to create issues, which would be appear in a "needs approval" column of the project board. Project leads could view and approve the issues (or request changes), adding them to a "approved" column of the board. Contributors can then assign themselves to issues in the "approved" column

Let me know what you think!

**Comment by Caleb-Cohen** [comment]
Number 1: Agreed, and that's already in progress on my behalf.
Number 2: I've taken a look at templates, and agree as well, but I need to delve a little further. 
Number 3: Love it, will implement.
Number 4: Can't contributors already create issues?

**Reply by MatthewBozin** [reply]
Re: number 4, ah that's true we can. Could implement an approval process though so it's clear to contributors which issues they can start working on

**Reply by Caleb-Cohen** [reply]
I've implemented all four. Can you see if you can create an issue and assign it to projects?

**Reply by MatthewBozin** [reply]
It doesn't look like I can assign myself to an issue, label it, or assign it to projects

**Reply by Caleb-Cohen** [reply]
> It doesn't look like I can assign myself to an issue, label it, or assign it to projects

I don't think this is accomplishable without making you a collaborator. Is that correct?

**Reply by ghost** [reply]
Not sure if you was looking for better ways to police the repo , but the first link was some options I figured you might have missed. 
      https://github.com/settings/code_review_limits

Now this link below here, that's probably Da Answer period
      https://github.com/WinstonHall/Together-Next-App/settings/access
